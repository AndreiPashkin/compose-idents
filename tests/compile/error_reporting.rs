//! The error message should correctly point to problematic location in the code-block and also
//! report source identifier and the substitution.
use compose_idents::compose_idents;

compose_idents!(my_fn = "foo bar", {
    fn my_fn() -> u32 {
        42
    }
},);

fn main() {}
