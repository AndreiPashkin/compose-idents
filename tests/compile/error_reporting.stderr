error: SubstitutionError: failed to substitute:

         my_fn

       with:

         "foo bar"

       Encountered an error:

         expected identifier
 --> tests/compile/error_reporting.rs:6:8
  |
6 |     fn my_fn() -> u32 {
  |        ^^^^^
